R/06-calculate_gi.R

```diff
@@ -33,39 +33,75 @@ calc_gi <- function(.data = NULL,
   ## calculate expected double-targeting GI score by summing the two mean single-targeting
   ## CRISPR scores for that paralog pair
   gi_calc_df <- gimap_dataset$crispr_score %>%
-    dplyr::mutate(expected_crispr = dplyr::case_when(
-      target_type == "gene_gene" ~ mean_single_target_crispr_1 + mean_single_target_crispr_2,
-      target_type == "gene_ctrl" ~ mean_single_target_crispr_1 + mean_double_control_crispr_2,
-      target_type == "ctrl_gene" ~ mean_double_control_crispr_1 + mean_single_target_crispr_2
-    ))
-  
-  # Calculating mean crisprs
-  overall_results <- gi_calc_df %>%
+    dplyr::mutate(expected_crispr_double = mean_single_target_crispr_1 + mean_single_target_crispr_2,
+                  expected_crispr_single_1 = mean_single_target_crispr_1 +  mean_double_control_crispr,
+                  expected_crispr_single_2 = mean_single_target_crispr_2 +  mean_double_control_crispr
+                  )
+
+  mean_double_expected_df <- gi_calc_df %>%
     dplyr::group_by(rep, pgRNA_target_double) %>%
     dplyr::summarize(
-      mean_expected_crispr = mean(expected_crispr, na.rm = TRUE),
-      mean_observed_crispr = mean(double_crispr_score, na.rm = TRUE)
-    ) %>%
-    group_modify(~ broom::tidy(lm(mean_observed_crispr ~ mean_expected_crispr, data = .x)))
+    mean_expected_double_crispr = mean(expected_crispr_double, na.rm = TRUE)
```

Comment:
Some of the stats needed adjusting because the single and double crispr scores are calculated differently in their own linear models. @kweave and I pair programmed on this together, referencing the original code.